pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -286,31 +286,46 @@ func (k *KSMCheck) Configure(senderManager sender.SenderManager, integrationConf
 	builder := kubestatemetrics.New()
 	builder.WithUsingAPIServerCache(k.instance.UseAPIServerCache)
 
-	// Due to how init is done, we cannot use GetAPIClient in `Run()` method
-	// So we are waiting for a reasonable amount of time here in case.
-	// We cannot wait forever as there's no way to be notified of shutdown
-	apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
-	defer apiCancel()
-	c, err := apiserver.WaitForAPIClient(apiCtx)
-	if err != nil {
-		return err
-	}
+	k.configurePodCollection(builder, k.instance.Collectors)
 
-	// Discover resources that are currently available
-	resources, err := discoverResources(c.Cl.Discovery())
-	if err != nil {
-		return err
-	}
+	var collectors []string
+	var apiServerClient *apiserver.APIClient
+	var resources []*v1.APIResourceList
 
-	// Prepare the collectors for the resources specified in the configuration file.
-	collectors, err := filterUnknownCollectors(k.instance.Collectors, resources)
-	if err != nil {
-		return err
-	}
+	switch k.instance.PodCollectionMode {
+	case nodeKubeletPodCollection:
+		// In this case we don't need to set up anything related to the API
+		// server.
+		collectors = []string{"pods"}
+	case defaultPodCollection, clusterUnassignedPodCollection:
+		// Due to how init is done, we cannot use GetAPIClient in `Run()` method
```

Comment:
For reviewers, just in case the diff is a bit difficult to read, the code in this `case` is the same as the one we had before.